cdap-api-common/src/main/java/io/cdap/cdap/api/exception/ErrorCategory.java

```diff
@@ -52,8 +55,24 @@ public ErrorCategory(ErrorCategoryEnum errorCategory, String subCategory) {
    * Returns the category of the error.
    */
   public String getErrorCategory() {
-    return errorCategory == null ? ErrorCategoryEnum.OTHERS.toString() : subCategory == null ?
-        errorCategory.toString() : String.format("%s-'%s'", errorCategory, subCategory);
+    return errorCategory == null ? ErrorCategoryEnum.OTHERS.toString() : subCategory == null
+        ? errorCategory.toString() : String.format("%s-'%s'", errorCategory, subCategory);
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (!(o instanceof ErrorCategory)) {
+      return false;
+    }
+
+    ErrorCategory that = (ErrorCategory) o;
+    return Objects.equals(this.errorCategory, that.errorCategory)
+        && Objects.equals(this.subCategory, that.subCategory);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(getErrorCategory());
```

Comment:
Please don't do format for hashCode either

Reply:
done.